games-engines/gemrb/gemrb-0.9.3.recipe

```diff
@@ -0,0 +1,119 @@
+SUMMARY="An Infinity Engine Emulator"
+DESCRIPTION="GemRB (Game Engine Made with preRendered Background) \
+is a new implementation of the original Infinity Engine \
+(used in Baldur's Gate, Icewind Dale, Planescape: Torment etc.) \
+to Linux/Unix, MacOS X, Windows (and Haiku) with some enhancements.
+
+(Originals games' data is not included)"
+HOMEPAGE="https://gemrb.org"
+COPYRIGHT="2003-2024 The GemRB Team"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/gemrb/gemrb/archive/refs/tags/v$portVersion.tar.gz"
+CHECKSUM_SHA256="71bb16a77b84cd873d5582af675357c0cb94bbc64d54fac2e3397b366ac2adde"
+SOURCE_FILENAME="v$portVersion.tar.gz"
+PATCHES="gemrb-$portVersion.patchset"
+ADDITIONAL_FILES="gemrb.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86"
+
+GLOBAL_WRITABLE_FILES="
+	settings/gemrb/GemRB.cfg keep-old
+	"
+USER_SETTINGS_FILES="
+	settings/gemrb directory
+	"
+
+PROVIDES="
+	gemrb$secondaryArchSuffix = $portVersion
+	cmd:extend2da.py$secondaryArchSuffix = $portVersion
+	cmd:gemrb$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libSDL2_mixer_2.0$secondaryArchSuffix
+	lib:libvorbis$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libfreetype$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libopenal$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+	devel:libSDL2_mixer_2.0$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:python3
+	"
+
+defineDebugInfoPackage gemrb$secondaryArchSuffix \
+	$binDir/gemrb
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+
+	cmake .. \
+		$cmakeDirArgs \
+		-Wno-dev \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DDATA_DIR=$dataDir/gemrb \
+		-DMAN_DIR=$manDir/man6 \
+		-DSYSCONF_DIR=$settingsDir/gemrb \
+		-DDOC_DIR=$docDir
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build
+	make install
+
+	# Remove extra cfg files and setup Haiku-specific config
+	rm $settingsDir/gemrb/GemRB.cfg.noinstall.sample
+	rm $settingsDir/gemrb/GemRB.cfg.sample
+
+	# Edit Cache path
+	# Issue: cannot write cache files in most paths, falling back to ~/gemrb/cache
+	#cachedir=$(finddir B_USER_CACHE_DIRECTORY)/gemrb
+	#sed -i "s,.\/Cache2/,${cachedir},g" $settingsDir/gemrb/GemRB.cfg
+	sed -i "s,.\/Cache2/,~\/gemrb/cache,g" $settingsDir/gemrb/GemRB.cfg
```

Comment:
If that code just makes the directories it's not enough for the cache issue, there's something more to it unfortunately.

It would solve the saves issue though so I will add that to the recipe :)

Reply:
Nah I get it, having the cache in home is not ideal and I'd rather have that sorted as well